testdata/jennies/rawtypes/disjunctions/GoJSONMarshalling/disjunctions/types_json_marshalling_gen.go

```diff
@@ -8,7 +8,7 @@ func (resource StringOrBool) MarshalJSON() ([]byte, error) {
 		return json.Marshal(resource.Bool)
 	}
 
-	return nil, nil
+	return []byte("null"), nil
```

Comment:
I feel like this should be an error instead of returning null

Reply:
Valid point. Returning null in these cases makes the generated code very lenient: it will marshal to "null" some fields that the schema indicated as required but were not provided with a value (default or otherwise).

I updated this PR to return an error instead, and I'll open another PR to address https://github.com/grafana/grafana-foundation-sdk/issues/50 specifically :)